content/content/Examples/stateless-app-mirror.md

```diff
@@ -0,0 +1,290 @@
+will need modify the commands below to reference the correct cluster context.
+
+# Deploy Guestbook Application in "source" Cluster
+
+You will be deploying
+[Kubernetes' stateless guestbook application](https://kubernetes.io/docs/tutorials/stateless-application/guestbook/)
+modified here to be consumable via kustomize.
+The guestbook application consisists of:
+
+* redis leader deployment and service
+* redis follower deployment and service
+* guestbook front-end deployment and service
+
+
+```bash
+kubectl --context src create namespace guestbook
+kustomize build github.com/konveyor/crane-runner/examples/resources/guestbook | kubectl --context src --namespace guestbook apply -f -
+kubectl --context src --namespace guestbook wait --for=condition=ready pod --selector=app=guestbook --timeout=180s
+```
+
+**Optional**
+
+Forward localhost traffic to the frontend of the guestbook application:
+
+```bash
+kubectl --context src --namespace guestbook port-forward svc/frontend 8080:80
+```
+
+Now you should be able to access the guestbook from the browser of your choice
+via `localhost:8080`.
+
+
+# Extract Resources From "source" Cluster
+
+Crane's `export` command is how you extract all of the resources you want from
+the "source" cluster.
+
+```bash
+crane export --context src --namespace guestbook
+```
+
+Make sure everything is working as expected by checking the `export` directory,
+it should look something like:
+
+```bash
+$ tree -a export
+export
+├── failures
+│   └── guestbook
+└── resources
+    └── guestbook
+        ├── ConfigMap_guestbook_kube-root-ca.crt.yaml
+        ├── Deployment_guestbook_frontend.yaml
+        ├── Deployment_guestbook_redis-master.yaml
+        ├── Deployment_guestbook_redis-slave.yaml
+        ├── Endpoints_guestbook_frontend.yaml
+        ├── Endpoints_guestbook_redis-master.yaml
+        ├── Endpoints_guestbook_redis-slave.yaml
+        ├── EndpointSlice_guestbook_frontend-bkqbs.yaml
+        ├── EndpointSlice_guestbook_redis-master-hxr5k.yaml
+        ├── EndpointSlice_guestbook_redis-slave-8wt7z.yaml
+        ├── Pod_guestbook_frontend-5fd859dcf6-5nvbm.yaml
+        ├── Pod_guestbook_frontend-5fd859dcf6-j8w94.yaml
+        ├── Pod_guestbook_frontend-5fd859dcf6-s9x8p.yaml
+        ├── Pod_guestbook_redis-master-55d9747c6c-6f9bz.yaml
+        ├── Pod_guestbook_redis-slave-5c6b4c5b47-jnrsr.yaml
+        ├── Pod_guestbook_redis-slave-5c6b4c5b47-xz776.yaml
+        ├── ReplicaSet_guestbook_frontend-5fd859dcf6.yaml
+        ├── ReplicaSet_guestbook_redis-master-55d9747c6c.yaml
+        ├── ReplicaSet_guestbook_redis-slave-5c6b4c5b47.yaml
+        ├── Secret_guestbook_default-token-5vsrb.yaml
+        ├── ServiceAccount_guestbook_default.yaml
+        ├── Service_guestbook_frontend.yaml
+        ├── Service_guestbook_redis-master.yaml
+        └── Service_guestbook_redis-slave.yaml
+
+4 directories, 24 files
+```
+
+**NOTE**
+
+What's happening here? `crane export` is using a discovery client to see all of
```

Comment:
Is it accurate to say it's using a "discovery client"? (I'm not actually sure if a client exists internally), or is it more accurate to call it the server's discovery API?

Reply:
I called it a discovery client based on https://github.com/konveyor/crane/blob/27d247cbbe960c38d5be331e9983d0192601560b/cmd/export/export.go#L96-L100

```
	discoveryClient, err := o.configFlags.ToDiscoveryClient()
	if err != nil {
		log.Errorf("cannot create discovery client: %#v", err)
		return err
	}
```

Reply:
+1 thought that might have been the case and I was unfamiliar. LGTM as is.